apps/editing-toolkit/editing-toolkit-plugin/starter-page-templates/class-starter-page-templates.php

```diff
@@ -104,6 +104,18 @@ public function register_meta_field() {
 			},
 		);
 		register_meta( 'post', '_starter_page_template', $args );
+
+		$args = array(
+			'type'           => 'string',
+			'description'    => 'Selected category',
+			'single'         => true,
+			'show_in_rest'   => true,
+			'object_subtype' => 'page',
+			'auth_callback'  => function () {
+				return current_user_can( 'edit_posts' );
+			},
+		);
+		register_meta( 'post', '_wp_layout_category', $args );
```

Comment:
I think we should call this `_wpcom_template_layout_category` or `_wpcom_template_layout_categories` (if we're allowing multiple templates to be added) to be more explicit and to make it clear that this is a WordPress.com-specific feature/value.
```suggestion
		register_meta( 'post', '_wpcom_template_layout_category', $args );
```

I am also thinking that we should validate against the set of pattern categories that exist on the server. I believe we can use code something like the following for that:
```php
				'sanitize_callback' => function ( $meta_value ) { // other fields are also available, see sanitize_meta() in wp-includes/meta.php
					if ( ! is_array( $meta_value ) ) {
						return array();
					}

					if ( ! class_exists( '\A8C\FSE\Starter_Page_Templates' ) ) {
						return array();
					}

					$starter_page_templates = \A8C\FSE\Starter_Page_Templates::get_instance();
					// We need to pass a locale in here, but we don't actually depend on it, so we use the default site locale to optimise hitting the pattern cache for the site.
					$all_page_templates     = $starter_page_templates->get_page_templates( $starter_page_templates->get_vertical_locale() );
					$all_categories = wp_list_pluck( $all_page_templates, 'categories' );
					// Not sure if we _need_ uniqueness, but it's going to be a smaller array.
					$unique_categories = array_unique(
						array_reduce(
							$all_categories,
							function ( $categories_for_template ) {
								return array_keys( $categories_for_template );
							}
						)
					);

					// Only permit values that are valid categories.
					return array_intersect( $meta_value, $unique_categories );
				},
```